testdata/default/cheats/StateDiffStructTest.t.sol

```diff
@@ -0,0 +1,307 @@
+        test.setNestedStruct(
+            111111111, // inner.value1
+            address(0xCAFE), // inner.addr
+            222222222, // inner.value2
+            333333333, // value
+            address(0xBEEF) // owner
+        );
+
+        // Get the state diff as JSON
+        string memory stateDiffJson = vm.getStateDiffJson();
+
+        // Debug: log the JSON for inspection
+        emit log_string("State diff JSON (testdata):");
+        emit log_string(stateDiffJson);
+
+        assertContains(
+            stateDiffJson,
+            '"decoded":{"previousValue":"0","newValue":"111111111"},"label":"nestedStruct.inner.value1"',
+            "Should decode inner.value1 correctly"
+        );
+
+        assertContains(
+            stateDiffJson,
+            '"decoded":{"previousValue":"0x0000000000000000000000000000000000000000","newValue":"0x000000000000000000000000000000000000cafE"},"label":"nestedStruct.inner.addr"',
+            "Should decode inner.addr correctly"
+        );
+
+        assertContains(
+            stateDiffJson,
+            '"decoded":{"previousValue":"0","newValue":"222222222"},"label":"nestedStruct.inner.value2"',
+            "Should decode inner.value2 correctly"
+        );
+
+        assertContains(
+            stateDiffJson,
+            "0x00000000000000000000000000000000000000000000000000000000069f6bc7",
+            "Slot 4 should contain inner.value1 in hex"
+        );
+
+        assertContains(
+            stateDiffJson,
+            '"decoded":{"previousValue":"0","newValue":"333333333"},"label":"nestedStruct.value"',
+            "Should decode nestedStruct.value correctly"
+        );
+
+        assertContains(
+            stateDiffJson,
+            '"decoded":{"previousValue":"0x0000000000000000000000000000000000000000","newValue":"0x000000000000000000000000000000000000bEEF"},"label":"nestedStruct.owner"',
+            "Should decode nestedStruct.owner correctly"
+        );
+
+        // Stop recording
+        vm.stopAndReturnStateDiff();
+    }
+
+    // Helper function to check if a string contains a substring
+    function assertContains(string memory haystack, string memory needle, string memory message) internal pure {
+        bytes memory haystackBytes = bytes(haystack);
+        bytes memory needleBytes = bytes(needle);
+
+        if (needleBytes.length > haystackBytes.length) {
+            revert(message);
+        }
+
+        bool found = false;
+        for (uint256 i = 0; i <= haystackBytes.length - needleBytes.length; i++) {
+            bool isMatch = true;
+            for (uint256 j = 0; j < needleBytes.length; j++) {
+                if (haystackBytes[i + j] != needleBytes[j]) {
+                    isMatch = false;
+                    break;
+                }
+            }
+            if (isMatch) {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            revert(message);
+        }
+    }
+}
```

Comment:
can we move this to a helper, this is duplicated in https://github.com/foundry-rs/foundry/pull/11381/files#diff-7c006befc48f0502d1653777375b064ac35b0866dc7fe26fb332e78a5edc9371